src/taskProvider.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as serverUtils from './omnisharp/utils';
+import * as vscode from './vscodeAdapter';
+
+import { OmniSharpServer } from './omnisharp/server';
+import { TaskRevealKind } from 'vscode';
+
+let taskProvider: vscode.Disposable | undefined;
+
+export function activate(_context: vscode.ExtensionContext, server: OmniSharpServer): void {
```

Comment:
Most of the other features are registered by the extension defined in `extension.ts`. Does the adapter pattern mean that we will have a separate extension class for every feature we want to test through the adapter?

Reply:
I'm open for comment on this one. I prefer smaller, task-oriented units of code as I find them easier to understand, test, and modify.